src/uct/base/uct_iface.h

```diff
@@ -353,11 +354,22 @@ typedef void (*uct_iface_mpool_init_obj_cb_t)(uct_iface_h iface, void *obj,
 
 
 /**
- * Base structure for private data held inside a pending request. Contains
- * a queue element so we can put this on a queue.
+ * Structure for private data held inside a pending request.
  */
 typedef struct {
-    ucs_queue_elem_t  queue;
+    union {
```

Comment:
can we remove this typedef and just let each transport define its own priv_t?

Reply:
Transport still can don't use this one and define its own prov_t like it implemented in wireup ep because the struct is not visible from UCP but i don't see any other reason don't reuse the common code.

Reply:
i see it's used in CM, TCP, and failed ep. All of them AFAIK don't need flags (only ud does). 
why need to change common code, and not just UD transport?

Reply:
@yosefe it's easier to control size of private data and valid usage for all TLs when it's in common structure, many TLs use arbiter or queue, flags is a part of API, it's better to have everything in single place even if the struct is used partially against hard type casting in every transport.

I removed extra initialization of flags.